ui/app/src/Main.elm

```diff
@@ -53,8 +59,32 @@ init location =
 
                 _ ->
                     nullFilter
+
+        apiUrl =
+            flags
+                |> Json.decodeValue (Json.field "externalUrl" Json.string)
+                |> Result.withDefault "http://localhost:9093"
+                |> (\x ->
+                        if String.endsWith "/" x then
+                            String.dropRight 1 x
+                        else
+                            x
+                   )
+                |> Api.makeApiUrl
     in
-        update (urlUpdate location) (Model initSilenceList initSilenceView initSilenceForm initAlertList route filter initStatusModel)
+        update (urlUpdate location)
+            (Model
+                initSilenceList
+                initSilenceView
+                initSilenceForm
+                initAlertList
+                route
+                filter
+                initStatusModel
+                baseUrl
+                apiUrl
```

Comment:
Why do we need both `apiUrl` and `baseUrl`? I can't think of a case where they are not the same.